instrumentation/spring/spring-batch-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/batch/item/ItemTracer.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.batch.item;
+
+import static io.opentelemetry.api.trace.Span.Kind.INTERNAL;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.tracer.BaseTracer;
+import java.lang.ref.Reference;
+import java.lang.ref.WeakReference;
+import java.util.Optional;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.springframework.batch.core.scope.context.ChunkContext;
+
+public class ItemTracer extends BaseTracer {
+  private static final ItemTracer TRACER = new ItemTracer();
+
+  public static ItemTracer tracer() {
+    return TRACER;
+  }
+
+  // item-level listeners do not receive chunk/step context as parameters
+  // fortunately the whole chunk always executes on one thread - in Spring Batch chunk is almost
+  // synonymous with a DB transaction
+  // that makes it quite safe to store the current chunk context in a thread local and use it in
+  // each item span
+
+  private final ThreadLocal<WeakReference<ChunkContext>> currentChunk = new ThreadLocal<>();
```

Comment:
The ThreadLocal cleanup seems easy to reason about, I'd probably skip the weak reference.

Reply:
Done!